### PR TITLE
Add some stuff

### PR DESCRIPTION
**Pull request by nav-github01001**
## Description

made managers more useful.
Also @TheUntraceable pls accept this pr
or i resign

## Checklist

- [ ] - All code changes have been tested, if any
- [ ] - The documentation has been updated accordingly for any code changes
- [ ] - This PR fixes an already existing issue
- [x] - This PR adds a new feature (new functions, methods, classes, parameters, etc.)
- [ ] - This PR fixes a bug
- [ ] - This PR is a breaking change (removes/renames functions, parameters etc.)
- [ ] - This PR is NOT a code change (changes Documentation files, README.md, CONTRIBUTING.md etc.)

### CODE REVIEW

File: EpikCord/managers/channel_manager.py [hunk 1]
```diff
@@ -4,12 +4,12 @@
     Optional
 )
 from .cache_manager import CacheManager
-from ..__init__ import TextBasedChannel, GuildChannel, GuildTextChannel, GuildNewsChannel, VoiceChannel, DMChannel, ChannelCategory, GuildStoreChannel, GuildNewsThread, GuildStageChannel
+from ..__init__ import Client, TextBasedChannel, GuildChannel, GuildTextChannel, GuildNewsChannel, VoiceChannel, DMChannel, ChannelCategory, GuildStoreChannel, GuildNewsThread, GuildStageChannel
```
Comment (TheUntraceable) [inline EpikCord/managers/channel_manager.py:7]: This import will cause circular imports

File: EpikCord/managers/channel_manager.py [hunk 2]
```diff
@@ -24,4 +24,11 @@ async def fetch(self, channel_id, *, skip_cache: Optional[bool] = False):
...
         self.channels.append(channel)
-        return channel
\ No newline at end of file
+        return channel
+
+    async def add(self,guild_id,data:dict):
+        channel_data = await self.client.http.post(f"guilds/{guild_id}/channels", data=data)
+        self.channels.append(channel_data)
+        
+    async def edit_pos(self,guild_id,data:dict):
```
Comment (TheUntraceable) [inline EpikCord/managers/channel_manager.py:33]: Use the entire name for the func, `edit_pos` isn't helpful and it's misleading and you should probably use Kwargs

File: EpikCord/managers/roles_manager.py [hunk 3]
```diff
@@ -3,8 +3,18 @@
...
         response = await self.guild.client.http.post(f"guilds/{self.guild.id}/roles", data=role_data) # Send it to the server
-        await self.roles.append(role_data) # Store it locally
\ No newline at end of file
+        await self.roles.append(role_data) # Store it locally
+    
+    async def delete(self,role_id:int):
+        response = await self.guild.client.http.delete(f"guilds/{self.guild.id}/roles/{role_id}")
+        #Todd: pop the local roles
+
+    async def edit_role_pos(self,data:dict):
```
Comment (TheUntraceable) [inline EpikCord/managers/roles_manager.py:16]: Same flaw as the `edit_pos` func for channels manager

**Comment by TheUntraceable** [comment]
> or i resign

Seriously? All that in a simple pull request? That won't have any impact on if I accept or not.

**Comment by nav-github01001** [comment]
> > or i resign
> 
> Seriously? All that in a simple pull request? That won't have any impact on if I accept or not. Bad code is bad code. I won't accept bad code.

I just joked! dont takle it seriously

File: EpikCord/managers/channel_manager.py [hunk 4]
```diff
@@ -30,5 +30,5 @@ async def add(self,guild_id,data:dict):
         channel_data = await self.client.http.post(f"guilds/{guild_id}/channels", data=data)
         self.channels.append(channel_data)
         
-    async def edit_pos(self,guild_id,data:dict):
+    async def edit_position(self,guild_id,data:dict):
```
Comment (TheUntraceable) [inline EpikCord/managers/channel_manager.py:33]: Should be part of `GuildChannel` and should just take in an `int` for the position and not use that endpoint but `/guilds/channel/ID/` iirc

File: EpikCord/managers/roles_manager.py [hunk 5]
```diff
@@ -13,7 +13,7 @@ async def delete(self,role_id:int):
         response = await self.guild.client.http.delete(f"guilds/{self.guild.id}/roles/{role_id}")
         #Todd: pop the local roles
 
-    async def edit_role_pos(self,data:dict):
+    async def edit_role_position(self,data:dict):
```
Comment (TheUntraceable) [inline EpikCord/managers/roles_manager.py:16]: Same was what I said before, also just name it `edit_position`

File: EpikCord/managers/channel_manager.py (hunk 4 above)
Comment (nav-github01001) [reply EpikCord/managers/channel_manager.py:33]: ok will do that

Comment (nav-github01001) [reply EpikCord/managers/channel_manager.py:33]: according to discord API, the endpoint u gave is nopt correct

Comment (TheUntraceable) [reply EpikCord/managers/channel_manager.py]: Send a PATCH to `/channels/ID`, I think I already covered this though.

File: EpikCord/__init__.py [hunk 6]
```diff
@@ -1398,6 +1398,26 @@ async def fetch_pinned_messages(self) -> List[Message]:
         data = await response.json()
         return [Message(self.client, message) for message in data]
 
+    async def edit_position(self, position:int,parent_id:str,lock_permissions:bool=False) -> None:
```
Comment (TheUntraceable) [inline EpikCord/__init__.py:1401]: Why is `parent_id` and `lock_permissions` relevant? Just use `self.id`

**Pull request closed without merging**